File: 0101-0200/0165.Compare_Version_Numbers.cpp
//0ms






class Solution {
public:
    int compareVersion(string version1, string version2) {   
        int i = 0, j = 0, n1 = version1.size(), n2 = version2.size(), num1 = 0, num2 = 0; 
        while(i<n1 || j<n2)
        {
            while(i<n1 && version1[i]!='.')
            {
                num1 = num1*10+(version1[i]-'0');
                i++;
            }
            while(j<n2 && version2[j]!='.')
            {
                num2 = num2*10+(version2[j]-'0');
                j++;
            }
            if(num1>num2)
            {
                return 1;
            }
            if(num1 < num2) 
            {
                return -1;
            }
            i++;
            j++;
            num1 = 0;
            num2 = 0;
        }
        return 0; 
    }
};






//1ms







class Solution {
public:
    int compareVersion(string version1, string version2) {
        int i = 0;
        int j = 0;
        int ver = 1;
        bool full1=false;bool full2=false;
        while (i < version1.size() || j < version2.size()) {
            string number1;
             int num1;
             if(i>=version1.size()){
                full1=true;
                num1=0;
             }
            while (i < version1.size() and version1[i] != '.') {
                number1.push_back(version1[i]);
                i++;
            }
           
            if(!full1){
                num1 = stoi(number1);
            }
            

            string number2;
            int num2;
            if (j >= version2.size()) {
                full2=true;
                num2 = 0;
            }
            while (j<version2.size() and version2[j] != '.') {
                number2.push_back(version2[j]);
                j++;
            }
            if(!full2){
                num2=stoi(number2);
            }
            if (num1 > num2) {
                return 1;
            } else if (num2 > num1) {
                return -1;
            }
            i++;
            j++;
        }
        return 0;
    }
};

File: 0101-0200/0165.Compare_Version_Numbers.java
//0ms






class Solution {
    public int compareVersion(String version1, String version2) {
        int i = 0, j = 0;
        while (i < version1.length() || j < version2.length()) {
            int num1 = 0, num2 = 0;
            while (i < version1.length() && version1.charAt(i) != '.') {
                num1 = num1 * 10 + (version1.charAt(i++) - '0');
            }
            while (j < version2.length() && version2.charAt(j) != '.') {
                num2 = num2 * 10 + (version2.charAt(j++) - '0');
            }
            if (num1 < num2) {
                return -1;
            }
            if (num1 > num2) {
                return 1;
            }
            i++;
            j++;
        }
        return 0;
    }
}







//1ms






class Solution {
    public int compareVersion(String version1, String version2) {
        String[] v1 = version1.split("\\.");
        String[] v2 = version2.split("\\.");
        for(int i=0;i<Math.max(v1.length,v2.length);i++){
            int num1 = i<v1.length ? Integer.parseInt(v1[i]) : 0;
            int num2 = i<v2.length ? Integer.parseInt(v2[i]) : 0;
            if(num1<num2){
                return -1;
            }
            if(num1>num2){
                return 1;
            }
        }
        return 0;
    }
}






//0ms






class Solution {
    public int compareVersion(String v1, String v2) {
        int n = v1.length();
        int m = v2.length();
        int i = 0, j = 0;
        while (i < n || j < m) {
            int num1 = 0;
            while (i < n && v1.charAt(i) != '.') {
                num1 = num1 * 10 + (v1.charAt(i) - '0');
                i++;
            }
            int num2 = 0;
            while (j < m && v2.charAt(j) != '.') {
                num2 = num2 * 10 + (v2.charAt(j) - '0');
                j++;
            }
            if (num1 < num2) return -1;
            if (num1 > num2) return 1;
            i++;
            j++;
        }
        return 0;
    }
}
